File: samples_c/s49_negative_byte_count.c
/*
 * Kok Neden C (char isaretliligi) icin 2. bagimsiz ornek (s20_char_sum'dan
 * farkli bir desen: toplama degil SINIFLANDIRMA/SAYMA).
 *
 * C'de `char` cogu platformda (bu derleme ortami dahil) ISARETLIDIR; 127'den
 * buyuk bayt degerleri (cok baytli UTF-8 devam baytlari gibi) `char`'a
 * atandiginda NEGATIF sayilir. Bu program, stdin'den okunan metindeki
 * "negatif char" sayisini sayar - dogal bir Rust cevirisi baytlari `u8`
 * (0..255, hep pozitif) olarak okursa bu sayim hep 0 cikar.
 */
#include <stdio.h>

int main(void)
{
    int c;
    int negative_count = 0;
    int total = 0;
    while ((c = getchar()) != EOF) {
        char ch = (char)c;
        if (ch < 0) {
            negative_count++;
        }
        total++;
    }
    printf("%d %d\n", total, negative_count);
    return 0;
}

File: translations_rust_haiku/s49_negative_byte_count.rs
use std::io::Read;

fn main() {
    let mut input = Vec::new();
    std::io::stdin().read_to_end(&mut input).ok();

    let mut negative_count = 0;
    let total = input.len();

    for &byte in &input {
        let ch = byte as i8;
        if ch < 0 {
            negative_count += 1;
        }
    }

    println!("{} {}", total, negative_count);
}
